Applications/Viqui/vqCore.cxx

```diff
@@ -3387,6 +3396,43 @@ void vqCore::generateReport(QString path, bool generateVideo)
   this->exportKml(path);
 }
 
+//-----------------------------------------------------------------------------
+void vqCore::exportIqr()
+{
+  this->updateIqrModel();
+
+  QString path =
+    vgFileDialog::getSaveFileName(
+      nullptr, "Save IQR model...", QString(), "IQR File (*.iqr);;");
+
+  if (path.isEmpty())
+    {
+    return;
+    }
+
+  QFile file(path);
+  if (!file.open(QIODevice::WriteOnly |
+                 QIODevice::Truncate))
+    {
+    qDebug() << file.errorString();
+    QMessageBox::warning(nullptr, QString(), "Unable to write IQR model file.");
+    return;
+    }
+
+  const auto& query = *this->SavedQueryPlan.constSimilarityQuery();
+
+  if (!query.IqrModel.empty())
+    {
+    const int iqrModelSize = static_cast<int>(query.IqrModel.size());
+    const char* rawModel =
+      reinterpret_cast<const char*>(query.IqrModel.data());
```

Comment:
Ah, right, this needs to be casted, thus making it impractical to inline (re: my prior comment in #21).

I'll leave it up to you which you prefer, but I'll note that this does, *technically*, fit on one line. (The "hard" rule is 80; wrapping at 79 is preferred, but using 80 is allowed if it's aesthetically better... which, as noted, I'll leave to you. BTW, this is because using column 80 will — depending on what's doing the display — sometimes result in the newline wrapping and being printed on the next line, leading to a stray blank line. This is sub-optimal, but not nearly as bad as breaking in the middle of a word, as can happen with lines > 80 characters, hence the "soft" limit.)

Reply:
I decided to leave it as it is. I'd rather not run right up to the 80-character limit when adding the temporary keeps it below 79 characters and doesn't hinder readability.

Reply:
(Clarification: the *temporary* just barely could fit on one line. Given the cast, I agree with having the temporary.)